### ISSUE TITLE
Rework reconnect to the server when it is full

### ISSUE THREAD

**Issue by BannZay** [ddnet/ddnet#2214]
If server is full we spam server with connect attempts, this way we are getting "Too many connections in a short time" in return.

Why would not we request server status before doing reconnect ? We could request its status till we get something not like N/N

**Comment by Learath2** [comment]
Possibly because you get a TOCTOU race condition in that case. The only
viable rework would be something like a join queue but that's extra state
for the server to hold for non established connections, thus an increased
attack surface

**Comment by BannZay** [comment]
@Learath2, true. But you do exactly the same (but manually). With exactly the same problem. I just suggesting to automate this process

Yes, queues is much better. But those will work with DDNet servers only. So these features are not mutually exclusive

**Comment by BannZay** [comment]
Correct me, but I see no drawbacks if we replace old implementation with proposed. We still could fail, but lets avoid most fails to not be punished by the server

Want to point out that I am talking about reconnects only (if very first attempt to connect has been failed)

**Comment by Learath2** [comment]
The reconnection timings are what I would fix, maybe let the server tell the client what speed is acceptable.
